Assets/Scripts/Lua/CLRBindings/LuaArenaStatus.cs

```diff
@@ -29,69 +29,71 @@ public class LuaArenaStatus {
     }
 
     public void MoveAndResize(float x, float y, int width, int height, bool movePlayer = true, bool immediate = false) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) {
-            if (immediate) ArenaManager.instance.MoveAndResizeImmediate(x, y, width, height, movePlayer);
-            else           ArenaManager.instance.MoveAndResize(x, y, width, height, movePlayer);
-        } else
-            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        AssertIsDefending();
+
+        if (immediate) ArenaManager.instance.MoveAndResizeImmediate(x, y, width, height, movePlayer);
+        else           ArenaManager.instance.MoveAndResize(x, y, width, height, movePlayer);
     }
 
     public void MoveToAndResize(float x, float y, int width, int height, bool movePlayer = true, bool immediate = false) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) {
-            if (immediate) ArenaManager.instance.MoveToAndResizeImmediate(x, y, width, height, movePlayer);
-            else           ArenaManager.instance.MoveToAndResize(x, y, width, height, movePlayer);
-        } else
-            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        AssertIsDefending();
+
+        if (immediate) ArenaManager.instance.MoveToAndResizeImmediate(x, y, width, height, movePlayer);
+        else           ArenaManager.instance.MoveToAndResize(x, y, width, height, movePlayer);
     }
 
     public void SetSprite(string filename) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) ArenaManager.instance.sprite.Set(filename);
-        else                                                                    UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        AssertIsDefending();
+        ArenaManager.instance.sprite.Set(filename);
     }
 
     public void SetColor(float[] color) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) ArenaManager.instance.sprite.color = color;
-        else                                                                    UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        AssertIsDefending();
+        ArenaManager.instance.sprite.color = color;
     }
 
     public void SetColor32(float[] color32) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) ArenaManager.instance.sprite.color32 = color32;
-        else                                                                    UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        AssertIsDefending();
+        ArenaManager.instance.sprite.color32 = color32;
+    }
+
+    private void AssertIsDefending() {
+        if (UIController.instance.GetState() != UIController.UIState.DEFENDING) {
+            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
```

Comment:
Since you moved this around, it's probably the best to make the error something less dumb.